osu.Game.Rulesets.Catch/Skinning/Legacy/LegacyCatchHitExplosion.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Rulesets.Catch.Objects;
+using osu.Game.Rulesets.Catch.UI;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Skinning;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Catch.Skinning.Legacy
+{
+    public class LegacyCatchHitExplosion : CatchHitExplosion
+    {
+        private readonly Sprite explosion1;
+        private readonly Sprite explosion2;
+
+        [Resolved]
+        private Bindable<Color4> objectColour { get; set; }
+
+        [Resolved(Name = "CatcherWidth")]
+        private Bindable<float> catcherWidth { get; set; }
+
+        [Resolved]
+        private Bindable<float> catchPosition { get; set; }
+
+        [Resolved]
+        private Bindable<JudgementResult> judgementResult { get; set; }
+
+        [Resolved]
+        private Bindable<PalpableCatchHitObject> hitObject { get; set; }
+
+        private const float catcher_margin = (1 - Catcher.ALLOWED_CATCH_RANGE) / 2;
+
+        public LegacyCatchHitExplosion()
+        {
+            Anchor = Anchor.CentreLeft;
+            Origin = Anchor.BottomCentre;
+            RelativeSizeAxes = Axes.Both;
+            Scale = new Vector2(0.40f, 0.40f);
+
+            InternalChildren = new Drawable[]
+            {
+                explosion1 = new Sprite
+                {
+                    Origin = Anchor.CentreLeft,
+                    Blending = BlendingParameters.Additive,
+                    Rotation = -90,
+                },
+                explosion2 = new Sprite
+                {
+                    Origin = Anchor.CentreLeft,
+                    Blending = BlendingParameters.Additive,
+                    Rotation = -90,
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(ISkinSource source)
+        {
+            explosion1.Texture = source.GetTexture("scoreboard-explosion-2");
```

Comment:
As above, these should come from the default legacy skin, rather than the user's skin. I'm not 100% sure we have a method of doing this yet, so may require a bit of further investigation. There's an issue open for the more general case (#8019) but this is an explicit "do not want" case, which is a bit different.

Out of curiosity, does this even work right now? No catch skin should include this sprite as it won't actually apply on osu-stable.

Reply:
> Out of curiosity, does this even work right now? No catch skin should include this sprite as it won't actually apply on osu-stable.

You're right about that, I've only been testing with the "classic" skin so it did have the files. I'll take a look at getting it to work as required, then.

Reply:
Unfortunately this might not be an easy one, and may need a solution to #8019 as a prerequisite, but see how you go.